### PR TITLE
Function manual deploy instructions

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

Add manual deployment instructions for Appwrite Functions. This has been requested several times.

Also fixes a lot of debt we have for old, inconsistent formatting. (After we decided bolds are for UI elements, while `<code>` is used like ` in MD)

## Test Plan
row-1-column-1
row-2-column-1
row-3-column-1
row-4-column-1
row-5-column-1
row-6-column-1

### CODE REVIEW

File: app/views/docs/functions.phtml [hunk 1]
```diff
@@ -503,14 +503,64 @@ static RuntimeResponse &main(const RuntimeRequest &req, RuntimeResponse &res) {
 
 <p><a href="/docs/server/functions#functionsCreateDeployment" rel="noopener"><i class="icon-angle-circled-right margin-start-negative-tiny margin-end-tiny"></i> You can also create new code deployments using the Appwrite server API</a></p>
 
+<h3>Manual Deployment</h3>
+<p>You can also upload your functions to be deployed using the Appwrite console. The example below shows a typical Node.JS function, but the same idea applies to any other language.</p>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:507]:
```suggestion
<p>You can also upload your functions to be deployed using the Appwrite console. The example below shows a simple Node.JS function, but the same idea applies to any other language.</p>
```
`typical` sound passive-aggresive to me. Might be a me-issue. Just a suggestion.

File: app/views/docs/functions.phtml [hunk 2]
```diff
@@ -503,14 +503,64 @@ static RuntimeResponse &main(const RuntimeRequest &req, RuntimeResponse &res) {
...
+
+<div class="ide margin-bottom">
+    <pre class="line-numbers"><code class="prism" data-prism>.
+├── package.json
+└── src
+    └── main.js
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:513]: We could simplify even more. No need for `src` folder. Only thing that would change is entrypoint from `src/main.js` to `main.js`.

Comment (Meldiron) [inline app/views/docs/functions.phtml:513]: I would stick to `index.js` as suggested name. Just a convention we use everywhere.

File: app/views/docs/functions.phtml [hunk 3]
```diff
@@ -503,14 +503,64 @@ static RuntimeResponse &main(const RuntimeRequest &req, RuntimeResponse &res) {
...
+</div>
+
+<p>First, package your code files into the <code>.tar.gz</code> format with the tar command:</p>
+
+<ul class="phases clear" data-ui-phases>
+    <li>
+        <h3>Unix</h3>
+
+        <div class="ide margin-bottom" data-lang="bash" data-lang-label="Bash">
+            <pre class="line-numbers"><code class="prism language-bash" data-prism>tar -zcvf code.tar.gz src package.json</code></pre>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:524]: Better command: `tar -czf code.tar.gz --exclude code.tar.gz .`
(gzip everything in current directory EXCEPT gzip itself).

File: app/views/docs/functions.phtml [hunk 4]
```diff
@@ -503,14 +503,64 @@ static RuntimeResponse &main(const RuntimeRequest &req, RuntimeResponse &res) {
...
+
+<div class="ide margin-bottom">
+    <pre class="line-numbers"><code class="prism" data-prism>.
+├── package.json
+└── src
+    └── main.js
+</code></pre>
+</div>
+
+<p>First, package your code files into the <code>.tar.gz</code> format with the tar command:</p>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:517]: We **MUST** make it clear where this command should run. Which is inside the folder. Not outside, not inside `src`.

We do some folder location explaining in migration guide. That might be a good place to copy some text from: https://appwrite.io/docs/upgrade

File: app/views/docs/functions.phtml [hunk 5]
```diff
@@ -503,14 +503,64 @@ static RuntimeResponse &main(const RuntimeRequest &req, RuntimeResponse &res) {
...
+    </li>
+</ul>
+
+<p>Next, navigate to your Appwrite console and upload the function.</p>
+
+<ol class="margin-top margin-bottom-large text-size-normal">
+  <li class="margin-bottom-small">Navigate to the function you want to deploy.</li>
+  <li class="margin-bottom-small">Click <i class="icon-plus"></i> <b>Create deployment</b>.</li>
+  <li class="margin-bottom-small">Select the <b>Manual</b> tab.</li>
+  <li class="margin-bottom-small">Input the entry point of your function under <b>Entrypoint</b>. For the example above, it would be <code>src/main.js</code>.</li>
```
Comment (Meldiron) [inline app/views/docs/functions.phtml:549]: Update this if changing structure from suggestion earlier.

File: app/views/docs/functions.phtml (hunk 2 above)
Comment (gewenyu99) [reply app/views/docs/functions.phtml:513]: 👀 @Meldiron Should we notify the design team? This isn't what we use in console

File: app/views/docs/functions.phtml (hunk 3 above)
Comment (gewenyu99) [reply app/views/docs/functions.phtml:524]: 😬 They're likely to catch stuff like .envs, node_modules, .vscode. all kinds of garbage like that.

Also they might run it in the wrong folder.

File: app/views/docs/functions.phtml (hunk 4 above)
Comment (gewenyu99) [reply app/views/docs/functions.phtml:517]: @Meldiron They never read these, which is why I tell them to specify the files so they'd never have this issue :P

**Review by christyjacob4 — APPROVED** [review]

File: app/views/docs/functions.phtml (hunk 2 above)
Comment (Meldiron) [reply app/views/docs/functions.phtml]: Ill make sure to discuss it, same inconsistency is there in Open Runtimes.

File: app/views/docs/functions.phtml (hunk 3 above)
Comment (Meldiron) [reply app/views/docs/functions.phtml]: Valid point. Feel free to revert to your previous solution of specifying files. We could ignore node_modules, but people could use same command for Python. Specifying feels fine. They shouldnt have too complex of a function anway. if they do, Appwrite CLI is a way to do it.

**Review by Meldiron — APPROVED** [review]
Manual deployment part LGTM

File: app/views/docs/functions.phtml (hunk 3 above)
Comment (gewenyu99) [reply app/views/docs/functions.phtml]: Resolving for now. This should be fine. If we see people wondering why their manual deployments don't work/take too long/have packages track, we fix after.

**Pull request merged**